### PR TITLE
Introduce koin

### LINKED ISSUE #5: Setting DI
# OverView
- Setting DI with [Koin](https://insert-koin.io/)

This is blocked from #3

### PR DESCRIPTION
**Pull request by ho2ri2s**
## TL;DR

Introduce Koin for Dependency Injection

- close #5 

## Usage

all is [this](https://insert-koin.io/)

1. Create Class/Interface What you want to inject.(Like a TestRepository and TestRepositoryImpl)  
2. supply the Class/Interface from writing  module.kt.(in repository module)    
```
val repositoryModule = module {
    single<TestRepository> { TestRepositoryImpl() }
}
```  
This code is to supply TestRepositoryImpl with TestRepository type  
3. inject by constructor injection or field injection
   - constractor injection  
```
// module.kt  in test module that belong to TestViewModel
val testModule = module{
  viewModel { TestViewModel( get() ) }
}
```    
```
class TestViewModel(private val repository: TestRepository): ViewModel(){}
```    
  - field injection  
    - inejct normal class  

```
private val repository by inject()  
```      

    - inject ViewModel     

```
private val viewModel by viewModel()
```   

## Notes

- レビューOKならPRにApproveお願いします.
- ２人以上Approveついたらセルフマージします.

### CODE REVIEW

**Review by b4tchkn — APPROVED** [review]
danda_GJiine6
